ReactWindows/ReactNative.Net46/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -274,6 +289,8 @@ public override string Name
 
         public override void OnReactInstanceDispose()
         {
+            _cancelSource.Cancel();
```

Comment:
If you use `CancellationDisposable` from System.Reactive, it only requires a `Dispose` call, and I think it will protect you when accessing the token (so you won't need the `ObjectDisposedException` catch block below).